### ISSUE TITLE
Change this to work without needing Firebase?

### ISSUE THREAD

**Issue by kendallb** [stape-io/microsoft-ads-offline-conversion-tag#2]
Zaraz has full support for Bing in such a way that mimics how the client side pixel normally sends data to Bing, so the need for Firebase to store credentials should not be necessary?

https://github.com/managed-components/bing/tree/main

As you can see all it needs is the Bing UET Tag ID and it can send all the eCommerce events over:

![image](https://github.com/stape-io/microsoft-ads-offline-conversion-tag/assets/535825/94761c24-26f6-4e11-92f7-3dbcd4f1faeb)

Is it possible to rework this tool to work without Firebase, or is there something I am missing here?

**Comment by kendallb** [comment]
I am considering forking this project and trying to see if I can build my own template in a similar fashion to how Zaraz does it.

My question though, is how do you actually debug the code in the template to make sure it works? Is there a way to fire a break point in Chrome or something when the code runs, so you can see what it is doing? For server side tags, I have no ideas how you would do that (perhaps you just debug it old school with logging?).

**Comment by Bukashk0zzz** [comment]
This tag uses Offline conversion tracking to track everything.

That means you can pass user email, phone number, name, etc., to Bing Ads.
Also, you can do all of this without cookies/browsers etc.

That's the whole idea of the tag -> Do not rely on Browser.
In our cases, we often send webhooks from CRM to sGTM and then to Bing Ads.

What Zaraz does it use cookies to store Bing click id and then send it to Bing Ads on purchase.
That requires a browser and does not allow you to send user PII data, which, from our experience, brings lower results in attribution.

Not sure that forking tag helps you in any way.
Basically, you need to write completely different tags that use different APIs etc. And sure that this is a good idea as it will have lower attribution and you will rely on not documented API.

**Comment by giovaniortolani** [comment]
Closing this issue because MS developed a proper CAPI integration, and this tag will likely become obsolete.

- https://stape.io/solutions/microsoft-uet-conversion-api-tag
- https://learn.microsoft.com/en-us/advertising/guides/uet-conversion-api-integration
